packages/client/libs/typed-event-emitter/index.ts

```diff
@@ -41,4 +41,20 @@ export class TypedEventEmitter<
 
 		return this;
 	}
+
+	/**
+	 * This creates a promise that you can use for a single event.
+	 * @param event A event name
+	 */
+	public promise<E2 extends E, P extends unknown[] = Parameters<T[E2]>>(
```

Comment:
### My Proposals
- `await client.wait("pincall");`
- `await client.waitFor("pinncall");`

how about this?

Reply:
I wasn't satisfied for this naming, so your proposal looks great.
I think `wait` is simple, but probably it needs discussion.